src/shipit_workflow/shipit_workflow/tasks.py

```diff
@@ -25,9 +25,14 @@
 }
 
 
+class UnsupportedFlavor(Exception):
+    def __init__(self, description):
+        self.description = description
+
+
 def find_decision_task_id(project, revision):
     decision_task_route = 'gecko.v2.{project}.revision.{revision}.firefox.decision'.format(
-         project=project, revision=revision)
+        project=project, revision=revision)
```

Comment:
Future enhancement: If we want to support promoting the builds from revision X while using the taskgraph code from revision Y, we'll want to `find_decision_task_id` off revision Y and add revision X as a `previous_graph_id` (`previous_graph_ids` will look like `[Y, X, ...]`). This means we'll use the parameters from revision Y, but the builds that we use to replace our taskIds will come from graph X. We'll need to make sure that `version` hasn't been bumped in between (we can pass in the revision X version), and that the buildids aren't mismatched.

That may mean we want a `taskgraph_revision` and a `build_revision`.

(Though it's certainly possible that, in the future, the taskgraph will be stable and we won't ever need this.)